### ISSUE TITLE
The loss doesn't decrease during training process

### ISSUE THREAD

**Issue by Norma7755** [QQiuyp/FIN#7]
Hi, your work is great. I'm recovering your work by training the model. My traing settings are almost the same as yours except the batch size. After 100 epochs, the traing log shows that the loss doesn't decrease and stays at around 2.5. The error rate also stays as 50%. The details are as the log file shows. Is this normal?
[256-64-1.log](https://github.com/QQiuyp/FIN/files/13485639/256-64-1.log)

**Comment by QQiuyp** [comment]
I'm very sorry for the inconvenience caused. Indeed, I retried training using the default settings. In multiple training attempts, I encountered a scenario where the visual loss decreased rapidly, but the accuracy loss did not, making it quite challenging to achieve a stable decrease in both losses simultaneously. I am trying to find a more stable training strategy.

Initially, I set:
- learning rate (lr) = 1e-4
- message weight = 1500
keeping other settings unchanged. Once the accuracy reached above 90%, I then reduced the message weight to 1. After stabilizing the training, it was possible to further improve PSNR by increasing the stego weight.

After changing to:
- lr = 1e-4
- message weight = 1500
- stego weight = 1,
this was my log.

23-12-05 15:18:17.135 - INFO: Learning rate: 0.0001
23-12-05 15:18:17.136 - INFO: Train epoch 0:   Loss: 375.9791 | Stego_Loss: 0.0069 | Message_Loss: 0.2506 | Stego_Psnr: 27.9391 |Error:0.5023 |
23-12-05 15:18:17.470 - INFO: TEST:   PSNR_STEGO: 30.4430 | Error: 0.4863 | 
23-12-05 15:18:39.506 - INFO: Learning rate: 0.0001
23-12-05 15:18:39.506 - INFO: Train epoch 1:   Loss: 375.3716 | Stego_Loss: 0.0022 | Message_Loss: 0.2502 | Stego_Psnr: 33.0549 |Error:0.4982 |
23-12-05 15:18:39.818 - INFO: TEST:   PSNR_STEGO: 33.8181 | Error: 0.4756 | 
23-12-05 15:19:01.667 - INFO: Learning rate: 0.0001
23-12-05 15:19:01.667 - INFO: Train epoch 2:   Loss: 375.1740 | Stego_Loss: 0.0012 | Message_Loss: 0.2501 | Stego_Psnr: 35.2928 |Error:0.4970 |
23-12-05 15:19:01.979 - INFO: TEST:   PSNR_STEGO: 37.3909 | Error: 0.5088 | 
23-12-05 15:19:24.251 - INFO: Learning rate: 0.0001
23-12-05 15:19:24.251 - INFO: Train epoch 3:   Loss: 375.1478 | Stego_Loss: 0.0003 | Message_Loss: 0.2501 | Stego_Psnr: 42.3379 |Error:0.5017 |
23-12-05 15:19:24.581 - INFO: TEST:   PSNR_STEGO: 41.1803 | Error: 0.5000 | 
23-12-05 15:19:46.434 - INFO: Learning rate: 0.0001
23-12-05 15:19:46.435 - INFO: Train epoch 4:   Loss: 374.9619 | Stego_Loss: 0.0003 | Message_Loss: 0.2500 | Stego_Psnr: 40.9427 |Error:0.5042 |
23-12-05 15:19:46.749 - INFO: TEST:   PSNR_STEGO: 40.0637 | Error: 0.4824 | 
23-12-05 15:20:08.658 - INFO: Learning rate: 0.0001
23-12-05 15:20:08.659 - INFO: Train epoch 5:   Loss: 374.8926 | Stego_Loss: 0.0003 | Message_Loss: 0.2499 | Stego_Psnr: 41.2675 |Error:0.5066 |
23-12-05 15:20:08.987 - INFO: TEST:   PSNR_STEGO: 38.4775 | Error: 0.5215 | 
23-12-05 15:20:31.215 - INFO: Learning rate: 0.0001
23-12-05 15:20:31.216 - INFO: Train epoch 6:   Loss: 374.7771 | Stego_Loss: 0.0004 | Message_Loss: 0.2499 | Stego_Psnr: 40.9492 |Error:0.5103 |
23-12-05 15:20:31.608 - INFO: TEST:   PSNR_STEGO: 42.1319 | Error: 0.5137 | 
23-12-05 15:20:53.426 - INFO: Learning rate: 0.0001
23-12-05 15:20:53.426 - INFO: Train epoch 7:   Loss: 374.1825 | Stego_Loss: 0.0006 | Message_Loss: 0.2495 | Stego_Psnr: 38.6908 |Error:0.5299 |
23-12-05 15:20:53.746 - INFO: TEST:   PSNR_STEGO: 32.7184 | Error: 0.5947 | 
23-12-05 15:21:15.851 - INFO: Learning rate: 0.0001
23-12-05 15:21:15.851 - INFO: Train epoch 8:   Loss: 293.9867 | Stego_Loss: 8.5027 | Message_Loss: 0.1903 | Stego_Psnr: 17.4995 |Error:0.7765 |
23-12-05 15:21:16.168 - INFO: TEST:   PSNR_STEGO: 15.4175 | Error: 0.9082 | 
23-12-05 15:21:38.396 - INFO: Learning rate: 0.0001
23-12-05 15:21:38.396 - INFO: Train epoch 9:   Loss: 150.6443 | Stego_Loss: 0.4935 | Message_Loss: 0.1001 | Stego_Psnr: 15.7198 |Error:0.9323 |
23-12-05 15:21:38.783 - INFO: TEST:   PSNR_STEGO: 16.1449 | Error: 0.9688 | 
23-12-05 15:22:01.351 - INFO: Learning rate: 0.0001
23-12-05 15:22:01.351 - INFO: Train epoch 10:   Loss: 96.6239 | Stego_Loss: 0.0919 | Message_Loss: 0.0644 | Stego_Psnr: 18.1950 |Error:0.9799 |
23-12-05 15:22:01.734 - INFO: TEST:   PSNR_STEGO: 20.3792 | Error: 0.9951 | 
23-12-05 15:22:24.343 - INFO: Learning rate: 0.0001
23-12-05 15:22:24.343 - INFO: Train epoch 11:   Loss: 37.2130 | Stego_Loss: 0.0333 | Message_Loss: 0.0248 | Stego_Psnr: 21.3750 |Error:0.9970 |
23-12-05 15:22:24.724 - INFO: TEST:   PSNR_STEGO: 21.7904 | Error: 1.0000 | 
23-12-05 15:22:47.043 - INFO: Learning rate: 0.0001
23-12-05 15:22:47.044 - INFO: Train epoch 12:   Loss: 10.8279 | Stego_Loss: 0.0348 | Message_Loss: 0.0072 | Stego_Psnr: 21.1279 |Error:0.9998 |
23-12-05 15:22:47.435 - INFO: TEST:   PSNR_STEGO: 21.1459 | Error: 1.0000 | 
23-12-05 15:23:09.679 - INFO: Learning rate: 0.0001
23-12-05 15:23:09.680 - INFO: Train epoch 13:   Loss: 6.3010 | Stego_Loss: 0.0402 | Message_Loss: 0.0042 | Stego_Psnr: 20.4914 |Error:0.9999 |
23-12-05 15:23:10.055 - INFO: TEST:   PSNR_STEGO: 20.6875 | Error: 1.0000 | 
23-12-05 15:23:32.404 - INFO: Learning rate: 0.0001
23-12-05 15:23:32.404 - INFO: Train epoch 14:   Loss: 4.6696 | Stego_Loss: 0.0461 | Message_Loss: 0.0031 | Stego_Psnr: 19.9028 |Error:0.9999 |
23-12-05 15:23:32.782 - INFO: TEST:   PSNR_STEGO: 20.4367 | Error: 1.0000 | 

I end the training here, and change message weight = 1 for further training.
